### ISSUE TITLE
update react-measure to 2.x

### ISSUE THREAD

**Issue by jshimko** [reactioncommerce/reaction#2592]
[`react-measure`](https://github.com/souporserious/react-measure) has some minor breaking changes in the 2.x update.  Specifically, the `children` prop [must be a function](https://github.com/souporserious/react-measure#children-proptypesfunc).  Passing in a React class component does not work anymore. Not really sure why they'd put this limitation in place, but they did and we can't update until we change that.

The only place it's used is in the media gallery component/containers:

https://github.com/reactioncommerce/reaction/blob/development/imports/plugins/core/ui/client/components/media/mediaGallery.js#L75-L96

https://github.com/reactioncommerce/reaction/blob/development/imports/plugins/core/ui/client/containers/mediaGalleryContainer.js#L159-L177

**Comment by jshimko** [comment]
We should also consider using [their HOC](https://github.com/souporserious/react-measure#withcontentrecttypes-hoc) - especially since we now have the new components API and that would allow third parties to override the UI without having to implement the `react-measure` logic.

**Comment by janus-reith** [comment]
Funny coincidence - I just did that yesterday in my plugin, as I used react-measure around a react-konva canvas which replaces MediaGallery, and getting the Dimensions after updating the component was not working on some browsers including Firefox and Edge.
react-measure 2.x fixed these issues, so I second that.

**Comment by janus-reith** [comment]
@kieckhafer Answer for the duplicate: https://github.com/reactioncommerce/reaction/issues/2792  Yep, although my mediaGallery works completely different, the underlying logic is still quite similar, as far as I remember these were the modifications I needed for `mediaGalleryContainer` to work with 2.0 without changing too much:

```
<Measure
        bounds
        onResize={(contentRect) => {
          this.setState({ dimensions: contentRect.bounds })
        }}
      >
      {({ measureRef }) =>
      <div ref={measureRef}>
```
  _<MediaGallery [etc...] />_
```
       </div>
      }
</Measure>
```

From my experience, 2.0 is way more solid in and cross-browser compatible.

**Comment by jshimko** [comment]
Resolved by #2745
